admin-dev/themes/new-theme/js/pages/product/edit/product-options-manager.ts

```diff
@@ -53,10 +53,24 @@ export default class ProductOptionsManager {
   private initShowPriceToggler(): void {
     new FormFieldToggler({
       disablingInputSelector: ProductMap.options.availableForOrderInput,
-      matchingValue: '0',
+      matchingValue: '1',
       disableOnMatch: true,
       targetSelector: ProductMap.options.showPriceSwitchContainer,
-      toggleType: ToggleType.visibility,
+      toggleType: ToggleType.availability,
+    });
+
+    $(ProductMap.options.availableForOrderInput).on('change', (event) => {
+      const availableForOrder = $(event.target).val();
+      console.log('availableForOrder', availableForOrder);
+
+      // Automatically sets sho_price to true when available_for_order is true
+      if (availableForOrder) {
+        const $switchOn = $(`${ProductMap.options.showPriceInput}[value="1"]`);
+        const $switchOff = $(`${ProductMap.options.showPriceInput}[value="0"]`);
+
+        $switchOn.prop('checked', true);
+        $switchOff.prop('checked', false);
+      }
     });
```

Comment:
switching can also be done using vanilla js like (accidentally we worked on same pr, so i just leave an example instead of pushing my changes just in case :smile: ):

```
  private switchShowPrice(switchOn: boolean): void {
    const showPriceInputs: NodeListOf<HTMLInputElement> = document.querySelectorAll(ProductMap.options.showPriceInput);

    showPriceInputs.forEach((input) => {
      // eslint-disable-next-line no-param-reassign
      input.checked = input.value === '1' ? switchOn : !switchOn;
    });
  }
```

Reply:
Oh good to know!! I guess it's worth modifying this so that we have an example with both jquery and vanilla js  code